packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,401 @@
+`nimble-rich-text-editor`. This decision was made due to its extensive range of customization options compared to other third-party libraries for rich
+text editing, making it the ideal choice for meeting our specific use cases. For more details, refer
+[Comments FE Library Decision](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/docs/design-documents/Platform/Comments/Comments-FE-Library-Decision.md&version=GBmaster&_a=preview)
+document. Some of the highlighted points are mentioned below:
+
+1. Includes all [basic functionalities](https://tiptap.dev/) like bold, italics, numbered and bulleted lists, etc.
+2. Includes support to work seamlessly inside the `shadow root`.
+3. [Tiptap](https://tiptap.dev/introduction) is built on top of ProseMirror and so has access to all the powerful APIs provided by
+   ProseMirror through [`@tiptap/pm`](https://tiptap.dev/guide/prosemirror).
+4. Includes support to render the content of the editor with markdown input.
+5. Includes support to retrieve the content of the editor as HTML and Markdown output (using
+   [prosemirror-markdown](https://github.com/ProseMirror/prosemirror-markdown)).
+6. Includes extensions to support [@mention](https://tiptap.dev/api/nodes/mention) functionality (Future scope).
+
+The `nimble-rich-text-editor` is initialized by creating an instance of the [Editor](https://tiptap.dev/api/editor#introduction) class from Tiptap's core
+library. With that, we have access to all the APIs exposed, by utilizing some of their extensions like
+[StarterKit](https://tiptap.dev/api/extensions/starter-kit) which is a collection of the most popular Tiptap extensions including bold, italics, and all
+other basic rich text formatting options. All these formatting options can also be accessed individually through Tiptap's
+[marks](https://tiptap.dev/api/marks) and [nodes](https://tiptap.dev/api/nodes).
+
+-   The term `Marks` in the Tiptap are formatting styles applied to specific portions of text, such as bold, italic, or underlined.
+-   `Nodes` in the Tiptap editor are the building blocks of the document structure, representing different types of content elements like paragraphs,
+    headings, lists, images, and more.
+
+The rich text content entered in the editor is converted to markdown output using
+[prosemirror-markdown](https://github.com/ProseMirror/prosemirror-markdown) serializer. Here is the reference for the supported formatting schema in the
+markdown based on [CommonMark](http://commonmark.org/) flavor:
+
+-   Bold - `**Bold**`
+-   Italics - `*Italics*`
+-   Numbered list - `1. Numbered list`
+-   Bulleted list - `* Bulleted list`
+
+The `nimble-rich-text-viewer` will be responsible for converting the input markdown string to HTML Fragments with the help of
+`prosemirror-markdown` parser, which is then converted to HTML string and rendered into the component to view all rich text content.
+
+### Prototype
```

Comment:
It'll be good to have a discussion sometime about the process for converting this prototype into production code. I don't think we want one big PR for the entire feature at once. Rather it would be good to start small with a vertical slice and incrementally add functionality. This can help ensure better code quality and test coverage.

No spec updates are needed for this, just want to put it on your radar as a future discussion point.

Reply:
Sure, we will plan to discuss this once we start with the implementation and make sure to have a more granular PR for the entire component development.